### PR TITLE
[Merged by Bors] - Add docs & example for SystemParam

### PR DESCRIPTION
**Pull request by woubuc**
It took me a little while to figure out how to use the `SystemParam` derive macro to easily create my own params. So I figured I'd add some docs and an example with what I learned.

- Fixed a bug in the `SystemParam` derive macro where it didn't detect the correct crate name when used in an example (no longer relevant, replaced by #1426 - see further)
- Added some doc comments and a short example code block in the docs for the `SystemParam` trait
- Added a more complete example with explanatory comments in examples

### CODE REVIEW

File: crates/bevy_ecs/macros/src/lib.rs [hunk 1]
```diff
@@ -363,6 +364,10 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
     let manifest = Manifest::new().unwrap();
     let path_str = if let Some(package) = manifest.find(|name| name == "bevy") {
         format!("{}::ecs", package.name)
+    } else if let Some(package) = manifest.find(|name| name == "bevy_internal") {
+        format!("{}::ecs", package.name)
+    } else if let Some(package) = manifest.find(|name| name == "bevy_ecs") {
+        package.name
```
Comment (jakobhellermann) [inline crates/bevy_ecs/macros/src/lib.rs:370]: this is already fixed with #1426

Comment (woubuc) [reply crates/bevy_ecs/macros/src/lib.rs:370]: Ah, I hadn't seen that one. I'll wait for it to get merged and then update my PR.

File: crates/bevy_ecs/src/system/system_param.rs [hunk 2]
```diff
@@ -5,6 +5,25 @@ use crate::{
...
 use std::{any::TypeId, marker::PhantomData, sync::Arc};
+
+/// A parameter that can be used in a system function
+///
+/// # Derive
+/// This trait can be derived.
+///
+/// ```
+/// use bevy_ecs::prelude::*;
+/// use bevy_ecs::SystemParam;
```
Comment (MinerSebas) [inline crates/bevy_ecs/src/system/system_param.rs:16]:
```suggestion
/// # use bevy_ecs::prelude::*;
/// # use bevy_ecs::SystemParam;
```
All other doc examples hide the use statements, so the same should be done here.

Comment (woubuc) [reply crates/bevy_ecs/src/system/system_param.rs]: I've hidden the prelude, but I kept the SystemParam one to show that it needs to be imported explicitly (i.e. it's not part of the prelude).

**Comment by woubuc** [comment]
I've fixed the conflicts from #1426 and replaced my local fix with the better fix from that commit. Should be good to go now.

**Comment by cart** [comment]
Looks good to me. Thanks!

**Pull request closed without merging**